src/BlockTypes/ProductQuery.php

```diff
@@ -321,6 +322,10 @@ function ( $carry, $item ) {
 	 * @return array
 	 */
 	private function get_stock_status_query( $stock_statii ) {
+		if ( empty( $stock_statii ) ) {
+			return array();
+		}
```

Comment:
Alright, I see what you mean. We need to decide what is the approach we take at least within this class. I don't have a super strong opinion. Given the conversation in https://github.com/woocommerce/woocommerce-blocks/pull/7796, what do you think of the middle-ground I proposed there?

Reply:
Just commented there, I will update this PR following what we do with yours.